### PR TITLE
Rename jrun package to javarun

### LINKED ISSUE #19: Potential name clash on PyPI
https://pypi.org/project/jrun/
I do not know if we need to care at all.

### PR DESCRIPTION
**Pull request by ctrueden**
This avoids a clash on PyPI with an [existing jrun project](https://pypi.org/project/jrun/).

The only change required for downstream callers should be to change:
```python
import jrun
```
To:
```python
from javarun import jrun
```
And then hopefully everything else is the same.

The plan is to publish this package to PyPI as `javarun`.

Closes #19.

### CODE REVIEW

**Comment by ctrueden** [comment]
One downside of doing only a partial rename like this: the existing `jrun` project also has a console_scripts endpoint, so a `jrun` shell script gets installed for the Jupyter notebook CLI execution feature.

@hanslovsky Do you think we should rename this console script to `javarun` also, to avoid clashing?

**Comment by hanslovsky** [comment]
If we want to avoid potential clashes entirely we should probably do that. I installed the `jrun` from PyPI and thiese are all installed files, including an executable `jrun` script in the `bin` directory:
```
$ ls $HOME/local/tmp/jrun-pypi
/home/hanslovskyp/local/tmp/jrun-pypi
/home/hanslovskyp/local/tmp/jrun-pypi/lib
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun/cli.py
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun/__init__.py
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun/__pycache__
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun/__pycache__/__init__.cpython-37.pyc
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun/__pycache__/cli.cpython-37.pyc
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/SOURCES.txt
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/PKG-INFO
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/top_level.txt
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/installed-files.txt
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/dependency_links.txt
/home/hanslovskyp/local/tmp/jrun-pypi/lib/python3.7/site-packages/jrun-0.1.1-py3.7.egg-info/entry_points.txt
/home/hanslovskyp/local/tmp/jrun-pypi/bin
/home/hanslovskyp/local/tmp/jrun-pypi/bin/jrun
```

There is two ways to go ahead, in my opinion:
 1. Only rename the package (but not the module or executable names). While this might cause clashes it seems that the likelihood is very small and thus this is a reasonable way forward.
 2. Rename everything, including the shell script. That way we can be certain that we will not have future clashes (at least with `jrun`).

I do think that anything in between (i.e. rename module but not shell script) is not an option. I personally favor (2) and based on this existing PR I figure you do the same.

**Comment by imagejan** [comment]
Would renaming to `runj` be a viable option? It would be equally concise as the original `jrun`.

**Comment by hanslovsky** [comment]
I do not have a strong opinion on this, would work for me.

**Comment by ctrueden** [comment]
How about `jgo`? I found only [this apparently dead project](https://github.com/thomasmodeneis/jgo) to conflict with it. Seems available on conda-forge and PyPI. It is even shorter and easier to type than `jrun`! @hanslovsky @imagejan If you like it, I'll rename it today and upload to PyPI.

**Comment by ctrueden** [comment]
I like `jgo` more. Are you OK with it? Or do you foresee any problems with that name?

**Comment by hanslovsky** [comment]
I am OK with it.

File: javarun/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .jrun import resolve_dependencies
```
Comment (hanslovsky) [inline javarun/__init__.py]: We could also import `jrun_main` here.
```suggestion
from .jrun import resolve_dependencies, jrun_main
```

**Comment by ctrueden** [comment]
Great, thanks! I pushed a new `jgo` branch, but I am still troubleshooting it—the `jgo` shell script endpoint does not work yet for some reason. Will fix, and then release first version of `jgo` to PyPI as soon as everything is working.

Closing this PR, since it is obsolete.

File: javarun/__main__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import javarun
+if __name__ == '__main__':
+    javarun.jrun.jrun_main()
```
Comment (hanslovsky) [inline javarun/__main__.py]: Would just be `javarun.jrun_main` if `jrun_main` is imported in `__init__`.
```suggestion
    javarun.jrun_main()
```

File: setup.py [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
+    long_description=javarun_long_description,
+    long_description_content_type='text/markdown',
+    license='Public domain',
     url='https://github.com/scijava/jrun',
-    packages=['jrun'],
+    packages=['javarun'],
     entry_points={
         'console_scripts': [
-            'jrun=jrun.jrun:jrun_main'
+            'jrun=javarun.jrun:jrun_main'
```
Comment (hanslovsky) [inline setup.py]: Would just be javarun:jrun_main if jrun_main is imported in __init__.
```suggestion
            'jrun=javarun:jrun_main'
```

File: README.md [hunk 4]
```diff
@@ -48,28 +48,44 @@ jrun --help
 
 ### The Python module
 
-The `jrun/jrun.py` module (not to be confused with the
+The `javarun/jrun.py` module (not to be confused with the
```
Comment (hanslovsky) [inline README.md]:
```suggestion
The `javarun` module (not to be confused with the
```
Use `jgo` instead if appropriate.

**Comment by ctrueden** [comment]
Thanks @hanslovsky. FYI, work proceeded on the new `jgo` branch, and I think addresses all your comments.

**Comment by ctrueden** [comment]
https://pypi.org/project/jgo/

**Pull request closed without merging**